src/book/02-system/04-other/03-nugan.mdx

```diff
@@ -0,0 +1,193 @@
+---
+section: System
+chapter: Other Projects
+title: NUgan
+description: A CycleGAN implementation for dataset generation.
+slug: /system/other/nugan
+---
+
+The NUbots robots must be able to detect object in their environments so they are able to play soccer. We use machine learning to detect where objects are in the world. To find out more about our vision system, head over to [the vision page](/system/nubots/vision).
+
+Machine learning often requires data to train on. Rather than take thousands of pictures and annotate them by hand, we use advanced techniques to generate the data. [NUpbr](/system/nupbr) is used to generate semi-synthetic images with corresponding segmentation masks. These datasets are limited by their synthetic nature. To try to bridge this gap, we use CycleGAN to generate realistic images from NUpbr images.
+
+## GANs
+
+A [Generative Adversarial Network (GAN)](https://arxiv.org/abs/1406.2661) is a machine learning technique developed in 2014 which aims to create new data based on the given training data. A GAN consists of a generator and a discriminator. The generator creates fake data and the discriminator classifies the data as real or fake.
+
+A [Cycle-Consistent Adversarial Network (CycleGAN)](https://arxiv.org/abs/1703.10593) performs unpaired image-to-image translation. CycleGAN is based on [image-to-image translation GAN (pix2pix)](https://arxiv.org/abs/1611.07004). The main difference between these two GAN models is that CycleGAN uses unpaired data while pix2pix uses paired data. CycleGANs translate some domain A to another domain B, and vice versa. It trains not only by using A and B as input to the A and B generators respectively, but it also uses the output of each generator as the input for the other generator. This is the cycle input. The CycleGAN also uses an identity mapping to stabilise the training. It will input a real A image into the B generator, which should give back the original real A input. This is also done for real B images into the A generator.
```

Comment:
"It trains not only by using A and B as input to the A and B generators respectively, but it also uses the output of each generator as the input for the other generator."

I had trouble following this sentence

Reply:
Hm... I will see about adding a diagram, and maybe trying to fix up the sentence...

Reply:
Let me know if it's easier to understand now @KipHamiltons

Reply:
Awesome! the diagrams are a great addition.